### ISSUE TITLE
"input" not equal to "frft and inverse frft result"

### ISSUE THREAD

**Issue by dirtycomputer** [tunakasif/torch-frft#11]
import torch
from torch_frft.frft_module import frft

torch.manual_seed(0)
x = torch.rand(100)
y = frft(frft(x, 0.3),-0.3)
print(y)
assert torch.allclose(x, y.real, atol=1e-5)

**Comment by tunakasif** [comment]
Hello @dirtycomputer, it is the expected behavior. As described in [`README.md`](https://github.com/tunakasif/torch-frft/blob/5ec81829eb41a3444898ae186e97cd019f3269da/README.md?plain=1#L14), the implementation is based on [`fracF.m`](http://www.ee.bilkent.edu.tr/~haldun/fracF.m). As you can verify, the following MATLAB snippet also would not generate equivalent `x` and `y`.

```matlab
x = rand(100, 1);
y = fracF(fracF(x, 0.3), -0.3));
```

This is because the fast implementation `frft()` ([`fracF.m`](http://www.ee.bilkent.edu.tr/~haldun/fracF.m)) is a sampled approximation, and you are providing possibly the worst input, a random signal. You would not have this issue with `dfrft()` ([`dFRT.m`](http://www.ee.bilkent.edu.tr/~haldun/dFRT.m)), since it provides exact discrete FRFT but with an increased computational cost.

```python
import torch
from torch_frft.dfrft_module import dfrft

torch.manual_seed(0)
x = torch.rand(100)
y = dfrft(dfrft(x, 0.3),-0.3)
assert torch.allclose(x, y.real, atol=1e-5)
```

For details, please refer to [_Digital computation of the fractional Fourier transform_](https://ieeexplore.ieee.org/document/536672) and [_The discrete fractional Fourier transform_](https://ieeexplore.ieee.org/document/839980) papers.

**Comment by dirtycomputer** [comment]
Thank U for your fast reply. This is a very meaningful research.

**Comment by dirtycomputer** [comment]
It seems work!
`import torch
from torch.fft import fft, ifft, fftshift
from torch_frft.dfrft_module import dfrft
from torch_frft.frft_module import frft, frft_shifted
import matplotlib.pyplot as plt

def dfrft_idfrft(x, frac):
    return dfrft(dfrft(x, frac),-frac).real

def fft_ifft(x):
    return ifft(fft(x)).real

n = 20
torch.manual_seed(0)
x = torch.rand(n)
y1 = dfrft_idfrft(x, 0.5)
y2 = fft_ifft(x)
print(y1-y2)


plt.plot(torch.arange(n),y1,'s-',color = 'r',label="frft")
plt.plot(torch.arange(n),y2,'o-',color = 'g',label="fft")
plt.xlabel("time")
plt.ylabel("value")
plt.legend(loc = "best")
plt.show()`

![image](https://github.com/tunakasif/torch-frft/assets/48406770/5653ba49-5517-4e94-9cd4-532e9a03cbe6)
